### PR TITLE
rosparam to dynamic-reconfigure

### PR DESCRIPTION
**Pull request by h-kamada**
* parameter config (rosparam    ->     dynamic reconfigure)

### CODE REVIEW

File: jsk_pcl_ros/src/resize_points_publisher_nodelet.cpp [hunk 1]
```diff
@@ -36,12 +41,20 @@ namespace jsk_pcl_ros
...
       ConnectionBasedNodelet::onInit();
       pnh_->param("use_indices", use_indices_, false);
-      pnh_->param("step_x", step_x_, 2);
-      NODELET_INFO("step_x : %d", step_x_);
-      pnh_->param("step_y", step_y_, 2);
-      NODELET_INFO("step_y : %d", step_y_);
       pnh_->param("not_use_rgb", not_use_rgb_, false);
       pub_ = advertise<sensor_msgs::PointCloud2>(*pnh_, "output", 1);
+      srv_ = boost::make_shared <dynamic_reconfigure::Server<Config> > (*pnh_);
+      dynamic_reconfigure::Server<Config>::CallbackType f =
```
Comment (garaemon) [inline jsk_pcl_ros/src/resize_points_publisher_nodelet.cpp]: Create srv_ right after `ConnectionBasedNodelet::onInit`

**Comment by h-kamada** [comment]
fixed

**Comment by wkentaro** [comment]
I have a question about handling parameters with rosparam and dynamic
reconfigure.
In the test stage, dynamic reconfigure is very useful to adjust the param
for the purpose.
but after the testing how can I fix the value? how can I set default value
to the param?

now I launches dynamic reconfigure node to set default value to the param
like::

  <node pkg="dynamic_reconfigure" type="dynparam" name="$(anon dyparam1)"
args="set_from_parameters /dilate_mask_image" >
    <param name="size" type="int" value="5" />
    <param name="iterations" type="int" value="2" />
  </node>


On Tue, Apr 21, 2015 at 3:41 PM, h-kamada <notifications@github.com> wrote:

> fixed
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jsk-ros-pkg/jsk_recognition/pull/852#issuecomment-94654638>
> .
>



-- 
和田 健太郎 / Kentaro Wada
http://wkentaro.com <http://www.wkentaro.com>

**Comment by garaemon** [comment]
just <param> and <rosparam> should work

2015年4月21日火曜日、Kentaro Wada<notifications@github.com>さんは書きました:

> I have a question about handling parameters with rosparam and dynamic
> reconfigure.
> In the test stage, dynamic reconfigure is very useful to adjust the param
> for the purpose.
> but after the testing how can I fix the value? how can I set default value
> to the param?
>
> now I launches dynamic reconfigure node to set default value to the param
> like::
>
> <node pkg="dynamic_reconfigure" type="dynparam" name="$(anon dyparam1)"
> args="set_from_parameters /dilate_mask_image" >
> <param name="size" type="int" value="5" />
> <param name="iterations" type="int" value="2" />
> </node>
>
>
> On Tue, Apr 21, 2015 at 3:41 PM, h-kamada <notifications@github.com
> <javascript:_e(%7B%7D,'cvml','notifications@github.com');>> wrote:
>
> > fixed
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <
> https://github.com/jsk-ros-pkg/jsk_recognition/pull/852#issuecomment-94654638
> >
> > .
> >
>
>
>
> --
> 和田 健太郎 / Kentaro Wada
> http://wkentaro.com <http://www.wkentaro.com>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jsk-ros-pkg/jsk_recognition/pull/852#issuecomment-94657069>
> .
>


-- 
✉︎ from iPhone

**Comment by wkentaro** [comment]
I didnt know that. thank you.

cc: @aginika

On Tue, Apr 21, 2015 at 5:26 PM, Ryohei Ueda <notifications@github.com>
wrote:

> just <param> and <rosparam> should work
>
> 2015年4月21日火曜日、Kentaro Wada<notifications@github.com>さんは書きました:
>
> > I have a question about handling parameters with rosparam and dynamic
> > reconfigure.
> > In the test stage, dynamic reconfigure is very useful to adjust the param
> > for the purpose.
> > but after the testing how can I fix the value? how can I set default
> value
> > to the param?
> >
> > now I launches dynamic reconfigure node to set default value to the param
> > like::
> >
> > <node pkg="dynamic_reconfigure" type="dynparam" name="$(anon dyparam1)"
> > args="set_from_parameters /dilate_mask_image" >
> > <param name="size" type="int" value="5" />
> > <param name="iterations" type="int" value="2" />
> > </node>
> >
> >
> > On Tue, Apr 21, 2015 at 3:41 PM, h-kamada <notifications@github.com
> > <javascript:_e(%7B%7D,'cvml','notifications@github.com');>> wrote:
> >
> > > fixed
> > >
> > > —
> > > Reply to this email directly or view it on GitHub
> > > <
> >
> https://github.com/jsk-ros-pkg/jsk_recognition/pull/852#issuecomment-94654638
> > >
> > > .
> > >
> >
> >
> >
> > --
> > 和田 健太郎 / Kentaro Wada
> > http://wkentaro.com <http://www.wkentaro.com>
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <
> https://github.com/jsk-ros-pkg/jsk_recognition/pull/852#issuecomment-94657069
> >
> > .
> >
>
>
> --
> ✉︎ from iPhone
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jsk-ros-pkg/jsk_recognition/pull/852#issuecomment-94700757>
> .
>



-- 
和田 健太郎 / Kentaro Wada
http://wkentaro.com <http://www.wkentaro.com>

**Comment by garaemon** [comment]
dynamic_reconfgure server
1. reads initial parameter from rosparam
2. if not available, uses default parameter written in cfg file
3. enables service API
4. rqt_reconfigure calls service API of the dynamic_reconfigure server
5. If it's updated, dynamic_reconfigure servers updates rosparam

**Comment by wkentaro** [comment]
I could not find that information from ros wiki, so I needed to try to
check the order of setting default value.
It helps me a lot. thank u

2015年4月21日火曜日、Ryohei Ueda<notifications@github.com>さんは書きました:

> dynamic_reconfgure server
> 1. reads initial parameter from rosparam
> 2. if not available, uses default parameter written in cfg file
> 3. enables service API
> 4. rqt_reconfigure calls service API of the dynamic_reconfigure server
> 5. If it's updated, dynamic_reconfigure servers updates rosparam
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jsk-ros-pkg/jsk_recognition/pull/852#issuecomment-94712156>
> .
>


-- 
和田 健太郎 / Kentaro Wada
http://wkentaro.com <http://www.wkentaro.com>

**Pull request merged**